cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletEndpointTest.java

```diff
@@ -0,0 +1,263 @@
+    _userTaskManager = new UserTaskManager(1000, 10, TimeUnit.HOURS.toMillis(6),
+        100, _mockTime, _mockUUIDGenerator);
+  }
+
+  private static class MockResult implements CruiseControlResponse {
+    public void discardIrrelevantResponse(CruiseControlParameters parameters) { }
+    public void writeSuccessResponse(CruiseControlParameters parameters, HttpServletResponse response) throws IOException {
+    }
+  }
+
+  private static Object[] inputRequestParams(UUID userTaskId, String clientId, String endPoint, Map<String, String[]> params,
+      boolean addToRequest, String methodType) {
+    return new Object[]{userTaskId, clientId, endPoint, params, addToRequest, methodType};
+  }
+
+  private static Object[] outputRequestInfo(HttpServletRequest mockHttpServletRequest) {
+    return new Object[]{mockHttpServletRequest};
+  }
+
+  private static Object[] inputCreateTaskParams(HttpServletRequest request, Integer taskIndex, Integer futureIndex) {
+    return new Object[]{request, taskIndex, futureIndex};
+  }
+
+  private static Object[] outputCreateTaskInfo(OperationFuture future) {
+    return new Object[]{future};
+  }
+
+  private static OperationFuture getFuture(int idx) {
+    return (OperationFuture) ((ArrayList<Object[]>) _populateUserTaskManagerOutput).get(idx)[0];
+  }
+
+  // Creates 6 requests below. These will create 5 entries of UserTaskInfo (request 5 and 6 have same id, so will be 1 UserTaskInfo)
+  private void initializeServletRequests(HttpSession mockHttpSession, UserTaskManager.UUIDGenerator mockUUIDGenerator) {
+    Collection<Object[]> allParams = new ArrayList<>();
+
+    allParams.add(inputRequestParams(null, "0.0.0.1", EndPoint.LOAD.toString(), EMPTY_PARAM, false, GET_METHOD));
+    allParams.add(inputRequestParams(null, "0.0.0.1", EndPoint.PROPOSALS.toString(), EMPTY_PARAM, false, GET_METHOD));
+    allParams.add(inputRequestParams(null, "0.0.0.2", EndPoint.REBALANCE.toString(), EMPTY_PARAM, false, POST_METHOD));
+    allParams.add(inputRequestParams(null, "0.0.0.3", EndPoint.REBALANCE.toString(), DIFF_PARAM, false, POST_METHOD));
+    allParams.add(inputRequestParams(repeatUUID, "0.0.0.1", EndPoint.REMOVE_BROKER.toString(), EMPTY_PARAM, false, POST_METHOD));
+    allParams.add(inputRequestParams(repeatUUID, "0.0.0.4", EndPoint.REMOVE_BROKER.toString(), EMPTY_PARAM, true, POST_METHOD));
+
+    for (Object[] params : allParams) {
+      HttpServletRequest mockHttpServletRequest = prepareTestRequest(mockHttpSession, params[0], params[1], params[2],
+          params[3], mockUUIDGenerator, params[4], params[5]);
+      _initializeServletRequestsOutput.add(outputRequestInfo(mockHttpServletRequest));
+    }
+  }
+
+  // Create 5 User Tasks. Note the 5th and 6th one have same user task id, thus count as 1.
+  private void populateUserTaskManager(HttpServletResponse mockHttpServletResponse, UserTaskManager userTaskManager) {
+    Collection<Object[]> allParams = new ArrayList<>();
```

Comment:
Would it be possible to make the following changes:
1) `Collection<Object[]>` -> `List<Object[]>`
2) Remove casting before `allParams` below.